rig-neo4j/src/vector_index.rs

```diff
@@ -0,0 +1,358 @@
+    post_vector_search_filter: Option<String>,
+}
+
+impl SearchParams {
+    /// Initializes a new `SearchParams` with default values.
+    pub fn new(filter: Option<String>) -> Self {
+        Self {
+            post_vector_search_filter: filter,
+        }
+    }
+
+    pub fn filter(mut self, filter: String) -> Self {
+        self.post_vector_search_filter = Some(filter);
+        self
+    }
+}
+
+impl Default for SearchParams {
+    fn default() -> Self {
+        Self::new(None)
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RowResultNode<T> {
+    score: f64,
+    element_id: i64,
+    node: T,
+}
+
+#[derive(Debug, Deserialize)]
+struct RowResult {
+    score: f64,
+    element_id: i64,
+}
+
+impl<M: EmbeddingModel + std::marker::Sync + Send> VectorStoreIndex for Neo4jVectorIndex<M> {
+    /// Get the top n nodes and scores matching the query.
+    ///
+    /// #### Generic Type Parameters
+    ///
+    /// - `T`: The type used to deserialize the result from the Neo4j query.
+    ///        It must implement the `serde::Deserialize` trait.
+    ///
+    /// #### Returns
+    ///
+    /// Returns a `Result` containing a vector of tuples. Each tuple contains:
+    /// - A `f64` representing the similarity score
+    /// - A `String` representing the node ID
+    /// - A value of type `T` representing the deserialized node data
+    ///
+    async fn top_n<T: for<'a> Deserialize<'a> + std::marker::Send>(
+        &self,
+        query: &str,
+        n: usize,
+    ) -> Result<Vec<(f64, String, T)>, VectorStoreError> {
+        let prompt_embedding = self.embedding_model.embed_document(query).await?;
+        let query = self.build_vector_search_query(prompt_embedding, true, n);
+
+        let rows = self.execute_and_collect::<RowResultNode<T>>(query).await?;
+
+        let results = rows
+            .into_iter()
+            .map(|row| (row.score, row.element_id.to_string(), row.node))
+            .collect::<Vec<_>>();
+
+        Ok(results)
+    }
+
+    /// Get the top n ids and scores matching the query. Runs faster than top_n since it doesn't need to transfer and parse
+    /// the full nodes and embeddings to the client.
+    async fn top_n_ids(
+        &self,
+        query: &str,
+        n: usize,
+    ) -> Result<Vec<(f64, String)>, VectorStoreError> {
+        let prompt_embedding = self.embedding_model.embed_document(query).await?;
+
+        let query = self.build_vector_search_query(prompt_embedding, false, n);
+
+        let rows = self.execute_and_collect::<RowResult>(query).await?;
+
+        let results = rows
+            .into_iter()
+            .map(|row| (row.score, row.element_id.to_string()))
+            .collect::<Vec<_>>();
+
+        Ok(results)
+    }
+}
+
+// ===============================
+// Utilities to print results from a vector search
+// ===============================
+
+#[cfg(feature = "display")]
+#[allow(dead_code)]
+pub mod display {
```

Comment:
I'd move in the examples directory, as it's really only useful there and not really part of the scope of the companion crate. See the examples in `rig-lancedb` for how to have modules in the `examples` folder.

Reply:
I thought it was useful to have a way to visualize nicely the query results. 

You don`t think it's relevant? Otherwise yes I'll remove the feature flags and move this to /examples